Docs: Update built-in rule help links

Eitan-Shteinberg opened the pull request:
Updates the helpUri for several built-in rules to point to the correct documentation file.

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!-- Please add an informative description that covers the changes made by the pull request. -->


---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] The pull request does not introduce breaking changes.

### [General Guidelines](../CONTRIBUTING.md)
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request is clear and informative.
- [ ] I have added myself to the 'assignees'.
- [ ] I have added 'linked issues' if relevant.

### [Testing Guidelines](../CONTRIBUTING.md#tests)
- [ ] Pull request includes test coverage for the included changes.

Eitan-Shteinberg committed:
Docs: Update built-in rule help links

Updates the helpUri for several built-in rules to point to the correct documentation file.

JohnathonMohr commented on src/Analyzer.Core/Rules/BuiltInRules.json:
```diff
@@ -807,7 +807,7 @@
             },
             {
               "path": "properties.retentionDays",
-              "equals": 0 // 0 == unlimited retention
```
I think the comment here was helpful. Can you keep it?

Eitan-Shteinberg committed:
Docs: Update rule descriptions and recommendations

Updates the descriptions and recommendations for several built-in rules to be more accurate and up-to-date.

Eitan-Shteinberg committed:
Fix: Correct typo in TA-000020

Corrects the spelling of 'Contributor' in the description and recommendation for rule TA-000020.

Eitan-Shteinberg committed:
Feat: Update TA-000025 with specific AKS versions

Updates the recommendation and evaluation logic for rule TA-000025 to recommend specific, supported AKS versions based on the current release calendar.

Eitan-Shteinberg committed:
Refactor: Revert TA-000034 to specific API version

Reverts the recommendation for TA-000034 to use the specific API version '2019-12-01' and clarifies the security reasoning in the description.
